<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MonitorTab</class>
 <widget class="QWidget" name="MonitorTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>700</height>
   </rect>
  </property>
  <property name="styleSheet">
   <string notr="true">
    QWidget {
      background-color: #1e1e1e;
      color: #cccccc;
    }
    QGroupBox {
      font-weight: bold;
      border: 1px solid #3c3c3c;
      border-radius: 5px;
      margin-top: 10px;
      padding-top: 10px;
      background-color: #252526;
    }
    QGroupBox::title {
      subcontrol-origin: margin;
      left: 10px;
      padding: 0 5px;
      color: #007acc;
    }
    QFrame.statCard {
      background-color: #252526;
      border: 1px solid #3c3c3c;
      border-radius: 8px;
      padding: 15px;
    }
    QProgressBar {
      border: 1px solid #3c3c3c;
      border-radius: 4px;
      text-align: center;
      background-color: #2d2d2d;
      color: white;
      font-weight: bold;
    }
    QProgressBar::chunk {
      border-radius: 3px;
    }
    QPushButton {
      background-color: #0e639c;
      color: white;
      border: none;
      padding: 8px 16px;
      border-radius: 4px;
      font-weight: bold;
    }
    QPushButton:hover {
      background-color: #1177bb;
    }
    QCheckBox {
      color: #cccccc;
    }
    QCheckBox::indicator {
      width: 18px;
      height: 18px;
    }
    QCheckBox::indicator:unchecked {
      border: 2px solid #555555;
      background-color: #2d2d2d;
      border-radius: 3px;
    }
    QCheckBox::indicator:checked {
      border: 2px solid #007acc;
      background-color: #007acc;
      border-radius: 3px;
    }
   </string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="headerLayout">
     <item>
      <widget class="QLabel" name="lblTitle">
       <property name="text">
        <string>Real-time Performance Monitor</string>
       </property>
       <property name="styleSheet">
        <string notr="true">font-size: 18px; font-weight: bold; color: #ffffff;</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="headerSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="chkAutoRefresh">
       <property name="text">
        <string>Auto Refresh</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnRefresh">
       <property name="text">
        <string>Refresh</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnExport">
       <property name="text">
        <string>Export Data</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="statsCardsLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <widget class="QFrame" name="cardFps">
       <property name="styleSheet">
        <string notr="true">
         QFrame {
           background-color: #252526;
           border: 1px solid #3c3c3c;
           border-radius: 8px;
         }
        </string>
       </property>
       <layout class="QVBoxLayout" name="fpsCardLayout">
        <item>
         <widget class="QLabel" name="lblFpsTitle">
          <property name="text">
           <string>FPS</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 12px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblFpsValue">
          <property name="text">
           <string>0.0</string>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 42px; font-weight: bold; color: #4caf50;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblFpsAvg">
          <property name="text">
           <string>Avg: 0.0</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 11px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="cardLatency">
       <property name="styleSheet">
        <string notr="true">
         QFrame {
           background-color: #252526;
           border: 1px solid #3c3c3c;
           border-radius: 8px;
         }
        </string>
       </property>
       <layout class="QVBoxLayout" name="latencyCardLayout">
        <item>
         <widget class="QLabel" name="lblLatencyTitle">
          <property name="text">
           <string>LATENCY</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 12px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblLatencyValue">
          <property name="text">
           <string>0.0 ms</string>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 42px; font-weight: bold; color: #2196f3;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblLatencyAvg">
          <property name="text">
           <string>Avg: 0.0 ms</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 11px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="cardThroughput">
       <property name="styleSheet">
        <string notr="true">
         QFrame {
           background-color: #252526;
           border: 1px solid #3c3c3c;
           border-radius: 8px;
         }
        </string>
       </property>
       <layout class="QVBoxLayout" name="throughputCardLayout">
        <item>
         <widget class="QLabel" name="lblThroughputTitle">
          <property name="text">
           <string>THROUGHPUT</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 12px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblThroughputValue">
          <property name="text">
           <string>0.0 TOPS</string>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 42px; font-weight: bold; color: #ff9800;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblThroughputMax">
          <property name="text">
           <string>Max: 26 TOPS</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 11px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="cardTemp">
       <property name="styleSheet">
        <string notr="true">
         QFrame {
           background-color: #252526;
           border: 1px solid #3c3c3c;
           border-radius: 8px;
         }
        </string>
       </property>
       <layout class="QVBoxLayout" name="tempCardLayout">
        <item>
         <widget class="QLabel" name="lblTempTitle">
          <property name="text">
           <string>TEMPERATURE</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 12px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblTempValue">
          <property name="text">
           <string>-- Â°C</string>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 42px; font-weight: bold; color: #e91e63;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblTempStatus">
          <property name="text">
           <string>Status: --</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888888; font-size: 11px;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="chartsLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupFpsChart">
       <property name="title">
        <string>FPS History</string>
       </property>
       <layout class="QVBoxLayout" name="fpsChartLayout">
        <item>
         <widget class="QFrame" name="frameFpsChart">
          <property name="minimumHeight">
           <number>200</number>
          </property>
          <property name="styleSheet">
           <string notr="true">
            QFrame {
              background-color: #1a1a1a;
              border: 1px solid #3c3c3c;
              border-radius: 4px;
            }
           </string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupLatencyChart">
       <property name="title">
        <string>Latency History</string>
       </property>
       <layout class="QVBoxLayout" name="latencyChartLayout">
        <item>
         <widget class="QFrame" name="frameLatencyChart">
          <property name="minimumHeight">
           <number>200</number>
          </property>
          <property name="styleSheet">
           <string notr="true">
            QFrame {
              background-color: #1a1a1a;
              border: 1px solid #3c3c3c;
              border-radius: 4px;
            }
           </string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupUtilization">
     <property name="title">
      <string>Resource Utilization</string>
     </property>
     <layout class="QVBoxLayout" name="utilizationLayout">
      <property name="spacing">
       <number>12</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="npuUtilLayout">
        <item>
         <widget class="QLabel" name="lblNpuUtil">
          <property name="text">
           <string>NPU Utilization:</string>
          </property>
          <property name="minimumWidth">
           <number>150</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressNpu">
          <property name="value">
           <number>0</number>
          </property>
          <property name="minimumHeight">
           <number>25</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QProgressBar::chunk { background-color: #4caf50; }</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblNpuValue">
          <property name="text">
           <string>0%</string>
          </property>
          <property name="minimumWidth">
           <number>50</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="memUtilLayout">
        <item>
         <widget class="QLabel" name="lblMemUtil">
          <property name="text">
           <string>Memory Usage:</string>
          </property>
          <property name="minimumWidth">
           <number>150</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressMem">
          <property name="value">
           <number>0</number>
          </property>
          <property name="minimumHeight">
           <number>25</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QProgressBar::chunk { background-color: #2196f3; }</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblMemValue">
          <property name="text">
           <string>0 MB</string>
          </property>
          <property name="minimumWidth">
           <number>50</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="powerUtilLayout">
        <item>
         <widget class="QLabel" name="lblPowerUtil">
          <property name="text">
           <string>Power Consumption:</string>
          </property>
          <property name="minimumWidth">
           <number>150</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressPower">
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="minimumHeight">
           <number>25</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QProgressBar::chunk { background-color: #ff9800; }</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblPowerValue">
          <property name="text">
           <string>0 W</string>
          </property>
          <property name="minimumWidth">
           <number>50</number>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
